File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int l = 0, r = nums.size()-1;
        while(l < r){
            if(nums[l] != 0){
                l++;
            } else{
                nums.erase(nums.begin()+l);
                nums.push_back(0);
            }
            if(nums[r] == 0){
                r--;
            }  
        }
    }
};

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        l = 0
        r = len(nums)-1
        while(l < r):
            if(nums[l] != 0):
                l+=1
            else: 
                nums.pop(l)
                nums.append(0)
            if(nums[r] == 0):
                r-=1
        